internal/integration/images_test.go

```diff
@@ -0,0 +1,225 @@
+			"cpu_architecture":  "arm64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/arm64/dependencies/rhcos/pre-release/latest/rhcos-live.aarch64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/arm64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.aarch64.img",
+		},
+		{
+			"openshift_version": "4.8",
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live-rootfs.x86_64.img",
+		},
+		{
+			"openshift_version": "pre-release",
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.x86_64.img",
+		},
+	}
+
+	assistedServer       *ghttp.Server
+	imageIDWithInitRD    = "bf25292a-dddd-49dc-ab9c-3fb4c1f07071"
+	imageIDWithoutInitRD = "bf25292a-dddd-49dc-ab9c-3fb4c1f07072"
+	ignitionContent      = []byte("someignitioncontent")
+	initrdContent        = []byte("someramdisk")
+	imageServer          *httptest.Server
+	imageClient          *http.Client
+
+	imageDir        string
+	scratchSpaceDir string
+	is              imagestore.ImageStore
+	ctxBg           = context.Background()
+)
+
+func verifyIgnition(fs filesystem.FileSystem) {
+	// TODO(djzager): Export isoeditor.ignitionImagePath?
+	f, err := fs.OpenFile("/images/ignition.img", os.O_RDONLY)
+	Expect(err).NotTo(HaveOccurred())
+
+	gzipReader, err := gzip.NewReader(f)
+	Expect(err).NotTo(HaveOccurred())
+	cpioReader := cpio.NewReader(gzipReader)
+	hdr, err := cpioReader.Next()
+	Expect(err).NotTo(HaveOccurred())
+	// TODO(djzager): Tie this to some constant?
+	Expect(hdr.Name).To(Equal("config.ign"))
+	Expect(hdr.Size).To(Equal(int64(len(ignitionContent))))
+
+	content, err := ioutil.ReadAll(cpioReader)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(content).To(Equal(ignitionContent))
+}
+
+func verifyRamdisk(fs filesystem.FileSystem, expectedContent []byte) {
+	// TODO(djzager): Should we export isoeditor.ramDiskImagePath?
+	f, err := fs.OpenFile("/images/assisted_installer_custom.img", os.O_RDONLY)
+	Expect(err).NotTo(HaveOccurred())
+
+	content, err := ioutil.ReadAll(f)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(bytes.TrimRight(content, "\x00")).To(Equal(expectedContent))
+}
+
+func getImage(path, imageType string, version map[string]string) filesystem.FileSystem {
+	resp, err := imageClient.Get(imageServer.URL + path)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+	isoPath := filepath.Join(scratchSpaceDir, fmt.Sprintf("test-%s-%s.%s.iso", version["openshift_version"], imageType, version["cpu_architecture"]))
```

Comment:
This could all be done with a tempfile for each call I think.
Then this function could return the path and you could get rid of `scratchSpaceDir`.